### ISSUE TITLE
Stream: enable creation from Queue or Channel

### ISSUE THREAD

**Issue by hearnadam** [getkyo/kyo#791]

**Comment by mobley-trent** [comment]
Hi @hearnadam since `kyo-prelude` can't depend on `kyo-core`, is it feasible to create a `common` module which will house the `Queue` object from `kyo-core` ? This module can then be a shared dependency in both `kyo-prelude` and `kyo-core`.

**Comment by johnhungerford** [comment]
I doubt we want to change the module dependency graph for this issue. If you're going to provide an integration between `Stream` and `Queue`, it should probably take the form of a method on `Queue`, e.g. `.toStream`.

**Comment by johnhungerford** [comment]
Now that we have `Channel` -> `Stream` integration, does it even make sense to try to stream directly from `Queue`? That will involve polling continuously, which strikes me as a bad idea. In order to implement a blocking poll, however, you'd essentially have to reimplement `Channel`.

**Comment by hearnadam** [comment]
@johnhungerford Good point. We can probably have `Queue` -> `Channel`, then convert that to a Stream. I will cut a PR.

**Comment by fwbrasil** [comment]
I think the new integration to produce a stream from a channel is all we need here. I can't see a good use case for promoting a queue to a channel and then a stream. It should start with channel directly.
